src/nc/gui/MainWindow.cpp

```diff
@@ -88,7 +88,8 @@ MainWindow::MainWindow(Branding branding, QWidget *parent):
 
     logToken_ = LogToken(logger);
 
-    settings_ = new QSettings(branding_.organizationName(), branding_.applicationName(), this);
+    settings_ = new QSettings(QString(branding.applicationName() + ".ini"), QSettings::IniFormat, this);
```

Comment:
Well, on Windows programs are usually installed to Program Files, which is not writable by usual users. And these programs are typically launched with Program Files/Program Name being their current directory. Nonetheless, usual users should be able to store settings.

If you want to allow the programmer to store options somewhere else, you can pass QSettings * to the MainWindow constructor. If nullptr is passed, the settings must be constructed the way they are now constructed. I would accept such a change.

Concerning the rest, I have not read everything yet. :)